packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -0,0 +1,186 @@
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/adb.dart';
+import 'package:shorebird_cli/src/bundletool.dart';
+import 'package:shorebird_cli/src/cache.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_validation_mixin.dart';
+import 'package:shorebird_cli/src/third_party/flutter_tools/lib/flutter_tools.dart';
+
+/// {@template preview_command}
+/// `shorebird preview` command.
+/// {@endtemplate}
+class PreviewCommand extends ShorebirdCommand
+    with ShorebirdConfigMixin, ShorebirdValidationMixin {
+  /// {@macro preview_command}
+  PreviewCommand({super.validators}) {
+    argParser
+      ..addOption(
+        'device-id',
+        help: 'The ID of the device or simulator to preview the release on.',
+      )
+      ..addOption(
+        'app-id',
+        help: 'The ID of the app to preview the release for.',
+      )
+      ..addOption(
+        'release-version',
+        help: 'The version of the release (e.g. "1.0.0").',
+      );
+  }
+
+  @override
+  String get name => 'preview';
+
+  @override
+  String get description => 'Preview a specific release on a device.';
+
+  @override
+  Future<int> run() async {
+    try {
+      await validatePreconditions(checkUserIsAuthenticated: true);
+    } on PreconditionFailedException catch (error) {
+      return error.exitCode.code;
+    }
+
+    Future<String?> promptForApp() async {
+      final apps = await codePushClientWrapper.getApps();
+      if (apps.isEmpty) return null;
+      final app = logger.chooseOne(
+        'Which app would you like to preview?',
+        choices: apps,
+        display: (app) => app.displayName,
+      );
+      return app.appId;
+    }
+
+    Future<String?> promptForReleaseVersion(String appId) async {
+      final releases = await codePushClientWrapper.getReleases(appId: appId);
+      if (releases.isEmpty) return null;
+      final release = logger.chooseOne(
+        'Which release would you like to preview?',
+        choices: releases,
+        display: (release) => release.version,
+      );
+      return release.version;
+    }
+
+    const platform = 'android';
+    final appId = results['app-id'] as String? ?? await promptForApp();
+
+    if (appId == null) {
+      logger.info('No apps found');
+      return ExitCode.success.code;
+    }
+
+    final releaseVersion = results['release-version'] as String? ??
+        await promptForReleaseVersion(appId);
+
+    if (releaseVersion == null) {
+      logger.info('No releases found');
+      return ExitCode.success.code;
+    }
+
+    final previewDirectory = cache.getPreviewDirectory(appId);
+    final aabPath = p.join(
+      previewDirectory.path,
+      '${platform}_$releaseVersion.aab',
+    );
+    final apksPath = p.join(
```

Comment:
TIL that `apks` was a thing

Reply:
It would probably be better to define this further down, closer to where it's first used (I think line 133)